runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -112,64 +82,64 @@ fn test_xtokens_recognize_multilocation() {
 			Box::new(
 				(Parent, Parachain(2), Junction::AccountId32 { network: Any, id: BOB }).into()
 			),
-			800_000_000
+			UnitWeightCost::get() * 4
 		));
-	});
-
-	// TODO::This mimic does not finialized yet. Still need mimic XcmRouter
-	// ParaB::execute_with(|| {
-	// 	XcmpQueue::on_idle(0, u64::MAX);
-	// 	assert_eq!(Tokens::free_balance(0, &AccountId::from(BOB)), 999_200_000_000);
-	// })
-}
-
-#[test]
-fn test_pallet_xcm_recognize_multilocation() {
-	TestNet::reset();
-	Relay::execute_with(|| {
-		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
-			&para_a_account(),
-			1_000_000_000_000,
+		assert_eq!(
+			Balances::free_balance(&AccountId::from(ALICE)),
+			500_000_000_000_000_000 - 1_000_000_000_000
 		);
-		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
-			&para_b_account(),
-			1_000_000_000_000,
+		assert_eq!(
+			Balances::free_balance(&sibling_account(2)),
+			1_000_000_000_000 /* Notice this is interesting, as it suggest local preserve XCM
+			                   * fee belongs to remote chain, not local chain */
 		);
 	});
-	ParaA::execute_with(|| {
-		register_channel_info(1, 2);
-	});
-	ParaB::execute_with(|| {
-		register_channel_info(2, 1);
-	});
+
 	ParaB::execute_with(|| {
-		assert_ok!(AssetManager::register_foreign_asset_type(
-			RawOrigin::Root.into(),
-			CurrencyId::ParachainReserve(Box::new(
-				(
-					Parent,
-					Parachain(1),
-					PalletInstance(<Balances as PalletInfoAccess>::index() as u8)
-				)
-					.into()
-			)),
-			Default::default()
-		));
-		assert_ok!(AssetManager::set_asset_units_per_second(
-			RawOrigin::Root.into(),
-			1,
-			1_000_000_000_000
+		assert_eq!(
+			Tokens::free_balance(
+				1, // Asset_id=1. The first registered Token in Para B
+				&AccountId::from(BOB)
+			),
+			u128::from(1_000_000_000_000 - UnitWeightCost::get() * 4)
+		);
+
+		// Send ParaA token back to ParachainA's BOB
+		assert_ok!(XTokens::transfer(
+			Origin::signed(AccountId::from(BOB)),
+			CurrencyId::ParachainReserve(Box::new(para_native_token_multilocation(1))),
+			500_000_000_000,
+			Box::new(
+				(Parent, Parachain(1), Junction::AccountId32 { network: Any, id: BOB }).into()
+			),
+			UnitWeightCost::get() * 4
 		));
 	});
 
+	ParaA::execute_with(|| {
+		assert_eq!(
+			Balances::free_balance(&AccountId::from(BOB)),
+			u128::from(500_000_000_000 - UnitWeightCost::get() * 4)
+		);
+		assert_eq!(
+			Balances::free_balance(&sibling_account(2)),
+			500_000_000_000 /* When assets transfer back, the xcm fee is moved to
+			                 * XcmFeesAccount, which is Treasury */
```

Comment:
> Thank you for many testcases, I didn't look into every single detail of them but most of them make sense to me 👍
> 
> Apart from the inline comments, there're a few extra questions/points from my side:
> 
> 1. Can we replace `AccountId::from(ALICE)` with `alice()` as it's already defined in setup? (same applies to bob)
> 2. Instead of `1_000_000_000_000`, can we use 1 * UNIT? As it increases the readability
> 3. The fact that this file `tests/xcm_parachain/mod.rs` actually contains all the xcm-tests is a bit confusing => I might prefer to have a file like `xcm_parachain.rs` and move the relayer_proof_builder to setup
> 4. I see both the XTokens version and raw version(= polkadot_xcm pallet) are tested, is it necessary? I mean if the XTokens interface is the **only** interface that we should call (as it's a wrapper to underlying xcms), then it's enough to just test xtokens => I might understand it wrongly though
> 5. As mentioned in the comments, `UnitWeightCost::get() * 4` is really used a lot across the file, maybe we should give it a variable to avoid repeating it. And it should only be used for one purpose (like dest_weight, but not as both dest_weight and fee)
> 6. Shall we have a testcase where non-registered assets can be handled correctly. I see there's a testcase for non-existing MultiLocation (`parachain(4)`), but I'm not sure if that's enough.

1. changed
2. make sense, changed
3. I have thought this too. Not only relay_sproof, but also methods like register_channel_info should integrate in setup. Will determine the specific solution in future.
4. I think it will be good to know the risk. If we want to manipulate our soverign account on other chains, we must allow something send xcm commands to certain degree of freedom, which means any xcm can be mimic. You can see the pallet_xcm's test is focus on execute and send methods. And I really think we should maintains these test.
5. There is actually no replacement, so if there are 4=n xcm commands, then it charge n times. It is just most call ending with 4 xcm command combo. I will add some comments around.
6.  That is tricky to do so, as it is otherparachains' freedom to select whatever multilocation they want. We may only test from if they are sibling chain, if they are relaychain etc.

Reply:
I add tests acccordingly. One update of my mistake. So non-native asset served as XCM fee will go to treasury, local native token will just get burned.

Reply:
I see, thanks.
Then the word `treasury` is a bit misleading as normally it's referred to the treasury account in the treasury pallet 😂

Reply:
check treasury account test added